### PR TITLE
added window exceed tests

### PR DESCRIPTION
**Pull request by dogusata**
This PR also includes the update of the golden screenshots taken for both chromium and webkit, which is 8 removal of the old webkit and addition of the 16 new ones. 24 files are related with the screenshots.

- Added new config to eslint to also check `./ui-tests` folder
- Added `chromium` for the test runs
- Added e2e tests as a step for PR check Github action
- Updated node version for the Github actions
- Updated eslint ignore list
- Updated tsconfig
- Updated animation wait helper (also to support page wide animations)
  - Added initial 500ms wait to let all animations and transitions states to settle
  - Grabbed all animation items (including transitions) on the main wrapper or given element selector and depending on the running animations count, added a check every 100ms to let them finish
- Added tab close test with middle click to tab label
- Added open new tab test with double click to tab bar
- Added window edge bleeding tests
- Added window resize tests
- Added prompt input's disabled state tests when an the screen is idle or in progress
- Updated screenshots folder according to the browser tested
- Updated theme with `fixed font` for consistency across platforms and OSs which the tests run
- Updated current tests with new wait for animation end helper
- Updated current tests' screenshots with new ones (by using as much as their target/parent elements)
  - **ie:** Follow up click results are only take the snap of the chats container now. In case in the future we change the prompt field block (footer etc.) or the tabs bar block, it will not change the look of the chats container and we won't need to update the snaps of them. For example, soon we'll change the remaining characters indicator on the prompt field which should not cause a visual change on the chat items container. This is why we're not taking a full page screenshot for the chat item or chat items container related tests.
- Added skip screen shots option to follow-up-click, close and open tab tests to be able to reuse them without same screenshots taken
- Updated `getSelector` method to give the `css=` prefix to the selectors **to avoid an issue** which we were not able to select an element with the same selector after it is selected for the first time, removed from screen and added back to the screen (not the same element, but the same selector). For example, after closing a tab and reopening a new one, it is not possible to select the **first tab** again. CSS selectors structure is just running at the time of the trigger in the runtime with default css selectors.
- Updated tests to use the new `getSelector` structure
- Changed default attribute for `testId` selectors from `mynah-ui-test-id` to `data-testid` for generic test tools usage. 
  - Now, when/if necessary, `testId`s can be directly used with `page.getByTestId` method of playwright.
- **Fixed** a potential accessibility issue on tabs bar (_thanks to e2e tests before the push_). The tab bar itself was getting a `disabled` attribute even if the `disabled` prop of the component is not set or is set to false. Since adding `''` empty string to an attribute will add the attribute.
- **Fixed** an issue for the button text size overrides (in some of the headless browsers and browsers run under linux)

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: ui-tests/__test__/main.spec.ts [hunk 1]
```diff
@@ -51,4 +51,16 @@ describe('Open MynahUI', () => {
...
+
+  it('should close the tab with middle click', async () => {
+    await closeTab(page, true, true);
+  });
+
+  it('should open a new tab with double click', async () => {
+    await openNewTab(page, true, true);
+  });
+
+  it('should keep the content inside window boundaries', async () => {
```
Comment (rahmaniaam) [inline ui-tests/__test__/main.spec.ts:63]: can we add test for when it correctly evaluate if contents is outside of boundary as well?

**Review by Jurredr — COMMENTED** [review]
LGTM

File: ui-tests/src/styles/theme.scss [hunk 2]
```diff
@@ -43,14 +109,14 @@
...
     --mynah-shadow-overlay: 0 7px 27px -12px rgba(0, 0, 0, 0.35);
-    --mynah-syntax-code-font-size: 1rem;
-    --mynah-syntax-code-line-height: 1.175rem;
-    --mynah-font-size-xxsmall: 0.75rem;
-    --mynah-font-size-xsmall: 0.85rem;
-    --mynah-font-size-small: 0.95rem;
-    --mynah-font-size-medium: 1rem;
-    --mynah-font-size-large: 1.125rem;
-    --mynah-line-height: 1.175rem;
+    --mynah-syntax-code-font-size: 13px;
```
[the remark this answers is not in the document]
Comment (Jurredr) [reply ui-tests/src/styles/theme.scss:112]: Was wondering this as well ^

File: src/styles/components/_button.scss [hunk 3]
```diff
@@ -7,6 +7,8 @@
     background-color: var(--mynah-color-button);
     color: var(--mynah-color-button-reverse);
     box-shadow: var(--mynah-shadow-button);
+    font-family: var(--mynah-font-family);
+    font-size: var(--mynah-font-size-medium) !important;
```
[the remark this answers is not in the document]
Comment (dogusata) [reply src/styles/components/_button.scss:11]: One of the tests suites on Linux overrides the button style in some certain case. Since they're OS dependent for elements, it is safe to give `!important` to avoid font-size overrides for button. And IDEs are using their own web renderers, we don't know how they can override.

File: ui-tests/__test__/main.spec.ts (hunk 1 above)
Comment (dogusata) [reply ui-tests/__test__/main.spec.ts:63]: Currently it already evaluates the boundaries.

File: ui-tests/src/styles/theme.scss (hunk 2 above)
[the remark this answers is not in the document]
Comment (dogusata) [reply ui-tests/src/styles/theme.scss:112]: Very nice question and good catch. The reason we're pushing it to fixed px sizes is to avoid OS based font rendering related **micro** pixel shifts. Even though you see everything exactly the same two screenshots, screenshot tool will not match them for different OSs. Screenshot tester catches some ghost pixels. Instead of letting the OS calculate the px, we're giving pixels. But it doesn't change the fact that we'll still use rem/em in necessary blocks.

**Review by rahmaniaam — APPROVED** [review]
